File: bddtests/src/main/kotlin/mocking/tpp/models/ViewPatientOverviewReply.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlElementWrapper
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(name = "ViewPatientOverviewReply")
@XmlAccessorType(XmlAccessType.FIELD)
data class ViewPatientOverviewReply(
        @XmlElementWrapper(name="DrugSensitivities")
        @field:XmlElement(name="Item")
        var drugSensitivities: MutableList<ViewPatientOverviewItem> = arrayListOf(),
        @XmlElementWrapper(name="Drugs")
        @field:XmlElement(name="Item")
        var drugs: MutableList<ViewPatientOverviewItem> = arrayListOf(),
        @XmlElementWrapper(name="PastRepeats")
        @field:XmlElement(name="Item")
        var pastRepeats: MutableList<ViewPatientOverviewItem> = arrayListOf(),
        @XmlElementWrapper(name="CurrentRepeats")
        @field:XmlElement(name="Item")
        var currentRepeats: MutableList<ViewPatientOverviewItem> = arrayListOf(),
        @XmlElementWrapper(name="Allergies")
        @field:XmlElement(name="Item")
        var allergies: MutableList<ViewPatientOverviewItem> = arrayListOf()
)

File: xamarinintegrationtests/NHSOnline.HttpMocks/Tpp/Models/ViewPatientOverviewReply.cs
using System.Collections.ObjectModel;
using System.Xml.Serialization;

namespace NHSOnline.HttpMocks.Tpp.Models
{
    public class ViewPatientOverviewReply
    {
        [XmlArray("DrugSensitivities")]
        [XmlArrayItem("Item", Type = typeof(ViewPatientOverViewItem))]
        public Collection<ViewPatientOverViewItem>? DrugSensitivities { get; }

        [XmlArray("Drugs")]
        [XmlArrayItem("Item", Type = typeof(ViewPatientOverViewItem))]

        public Collection<ViewPatientOverViewItem>? Drugs { get; set; } = new Collection<ViewPatientOverViewItem>();

        [XmlArray("PastRepeats")]
        [XmlArrayItem("Item", Type = typeof(ViewPatientOverViewItem))]
        public Collection<ViewPatientOverViewItem>? PastRepeats { get; set; } =
            new Collection<ViewPatientOverViewItem>();

        [XmlArray("CurrentRepeats")]
        [XmlArrayItem("Item", Type = typeof(ViewPatientOverViewItem))]
        public Collection<ViewPatientOverViewItem>? CurrentRepeats { get; set; } =
            new Collection<ViewPatientOverViewItem>();

        [XmlArray("Allergies")]
        [XmlArrayItem("Item", Type = typeof(ViewPatientOverViewItem))]
        public Collection<ViewPatientOverViewItem>? Allergies { get; set; } = new Collection<ViewPatientOverViewItem>();
    }
}